h2/settings.py

```diff
@@ -8,58 +8,139 @@
+#:    <h2.settings.SettingCodes.ENABLE_PUSH>`.
+ENABLE_PUSH = SettingCodes.ENABLE_PUSH
 
 #: Indicates the maximum number of concurrent streams that the sender will
 #: allow.
-MAX_CONCURRENT_STREAMS = SettingsFrame.MAX_CONCURRENT_STREAMS
+#:
+#: .. deprecated:: 2.6.0
+#:    Deprecated in favour of :class:`SettingCodes.MAX_CONCURRENT_STREAMS
+#:    <h2.settings.SettingCodes.MAX_CONCURRENT_STREAMS>`.
+MAX_CONCURRENT_STREAMS = SettingCodes.MAX_CONCURRENT_STREAMS
 
 #: Indicates the sender's initial window size (in octets) for stream-level flow
 #: control.
-INITIAL_WINDOW_SIZE = SettingsFrame.INITIAL_WINDOW_SIZE
+#:
+#: .. deprecated:: 2.6.0
+#:    Deprecated in favour of :class:`SettingCodes.INITIAL_WINDOW_SIZE
+#:    <h2.settings.SettingCodes.INITIAL_WINDOW_SIZE>`.
+INITIAL_WINDOW_SIZE = SettingCodes.INITIAL_WINDOW_SIZE
 
 #: Indicates the size of the largest frame payload that the sender is willing
 #: to receive, in octets.
-MAX_FRAME_SIZE = None
+#:
+#: .. deprecated:: 2.6.0
+#:    Deprecated in favour of :class:`SettingCodes.MAX_FRAME_SIZE
+#:    <h2.settings.SettingCodes.MAX_FRAME_SIZE>`.
+MAX_FRAME_SIZE = SettingCodes.MAX_FRAME_SIZE
 
 #: This advisory setting informs a peer of the maximum size of header list that
 #: the sender is prepared to accept, in octets.  The value is based on the
 #: uncompressed size of header fields, including the length of the name and
 #: value in octets plus an overhead of 32 octets for each header field.
 #:
-#: .. versionadded:: 2.5.0
-MAX_HEADER_LIST_SIZE = None
+#: .. deprecated:: 2.6.0
+#:    Deprecated in favour of :class:`SettingCodes.MAX_HEADER_LIST_SIZE
+#:    <h2.settings.SettingCodes.MAX_HEADER_LIST_SIZE>`.
+MAX_HEADER_LIST_SIZE = SettingCodes.MAX_HEADER_LIST_SIZE
+
+
+class ChangedSetting:
 
-try:  # Platform-specific: Hyperframe < 4.0.0
-    MAX_FRAME_SIZE = SettingsFrame.SETTINGS_MAX_FRAME_SIZE
-except AttributeError:  # Platform-specific: Hyperframe >= 4.0.0
-    MAX_FRAME_SIZE = SettingsFrame.MAX_FRAME_SIZE
+    def __init__(self, setting, original_value, new_value):
+        #: The setting code given. Either one of :class:`SettingCodes
+        #: <h2.settings.SettingCodes>` or ``int``
+        self.setting = setting
 
-try:  # Platform-specific: Hyperframe < 4.0.0
-    MAX_HEADER_LIST_SIZE = SettingsFrame.SETTINGS_MAX_HEADER_LIST_SIZE
-except AttributeError:  # Platform-specific: Hyperframe >= 4.0.0
-    MAX_HEADER_LIST_SIZE = SettingsFrame.MAX_HEADER_LIST_SIZE
+        #: The original value before being changed.
+        self.original_value = original_value
 
+        #: The new value after being changed.
+        self.new_value = new_value
 
-#: A value structure for storing changed settings.
-ChangedSetting = collections.namedtuple(
-    'ChangedSetting', ['setting', 'original_value', 'new_value']
-)
+    def __repr__(self):
+        return ("ChangedSetting(setting=%s, original_value=%s, "
+        "new_value=%s)") % (
+            self.setting.name,
```

Comment:
Well, I mean: it definitely won't work for integers, right? :grin:

Rather than isinstance, are you happy to just tolerate the default repr of an intenum member?